### PR TITLE
Add support for sparse depth maps

### PR DESCRIPTION
**Pull request by yimingzhou1**
# Background

In nerfstudio, the losses for depth don't support sparse depth maps.  

# Description

1. Modified `ds_nerf_depth_loss`, `urban_radiance_field_depth_loss` in `losses.py`.

2. Modified the way of computing `depth_mse` in eval. 

3. Fixed typo `Depht` -> `Depth`.

# Verification
Train `depth-nerfacto` with the `Replica` Dataset using only two views with sparse depth supervision (randomly masking out 90% of the depth map). left: groundtruth sparse depth map. right: predicted dense depth map. Results below are generated using `ds_nerf_depth_loss`. 

Screenshot 2023-02-07 at 2 05 44 PM

Screenshot 2023-02-07 at 2 06 34 PM

### CODE REVIEW

File: nerfstudio/model_components/losses.py [hunk 1]
```diff
@@ -226,7 +227,7 @@ def ds_nerf_depth_loss(
         Depth loss scalar.
     """
     loss = -torch.log(weights + EPS) * torch.exp(-((steps - termination_depth[:, None]) ** 2) / (2 * sigma)) * lengths
-    return torch.mean(loss.sum(-2))
+    return torch.mean((loss * depth_mask[:, None]).sum(-2))
```
Comment (jeffreyhuparallel) [inline nerfstudio/model_components/losses.py]: Can you avoid editing the function signature of this loss function? That makes the change more minimal.
```
loss = loss * (termination_depth > 0)[:, None]
```

File: nerfstudio/model_components/losses.py [hunk 2]
```diff
@@ -248,17 +250,18 @@ def urban_radiance_field_depth_loss(
...
     target_distribution = torch.distributions.normal.Normal(0.0, sigma / URF_SIGMA_SCALE_FACTOR)
     termination_depth = termination_depth[:, None]
     line_of_sight_loss_near_mask = torch.logical_and(
         steps <= termination_depth + sigma, steps >= termination_depth - sigma
     )
+    line_of_sight_loss_near_mask = torch.logical_and(line_of_sight_loss_near_mask, depth_mask[:, None])
     line_of_sight_loss_near = (weights - torch.exp(target_distribution.log_prob(steps - termination_depth))) ** 2
     line_of_sight_loss_near = (line_of_sight_loss_near_mask * line_of_sight_loss_near).sum(-2)
-    line_of_sight_loss_empty_mask = steps < termination_depth - sigma
+    line_of_sight_loss_empty_mask = torch.logical_and(steps < termination_depth - sigma, depth_mask[:, None])
```
Comment (jeffreyhuparallel) [inline nerfstudio/model_components/losses.py]: It is difficult for me to judge the correctness. Can you just mask the loss like in ds_nerf_loss?

File: nerfstudio/models/depth_nerfacto.py [hunk 3]
```diff
@@ -118,7 +120,7 @@ def get_image_metrics_and_images(
             far_plane=torch.max(ground_truth_depth),
         )
         images["depth"] = torch.cat([ground_truth_depth_colormap, predicted_depth_colormap], dim=1)
-        metrics["depth_mse"] = torch.nn.functional.mse_loss(outputs["depth"], ground_truth_depth)
+        metrics["depth_mse"] = torch.nn.functional.mse_loss(outputs["depth"]*(ground_truth_depth>0), ground_truth_depth)
```
Comment (jeffreyhuparallel) [inline nerfstudio/models/depth_nerfacto.py]: You need to mask the ground truth as well. Otherwise the zeros will always be correct and your metric will be biased high.

**Review by jeffreyhuparallel — CHANGES_REQUESTED** [review]

File: nerfstudio/models/depth_nerfacto.py (hunk 3 above)
Comment (jeffreyhuparallel) [reply nerfstudio/models/depth_nerfacto.py]: ```mse_loss(outputs["depth"][depth_mask], ground_truth_depth[depth_mask])```
